bioio/bio_image.py

```diff
@@ -264,25 +293,28 @@ def __init__(
         fs_kwargs: Dict[str, Any] = {},
         **kwargs: Any,
     ):
-        if reader is None:
-            # Determine reader class and create dask delayed array
-            self._plugin = BioImage.determine_plugin(
-                image, fs_kwargs=fs_kwargs, use_plugin_cache=use_plugin_cache, **kwargs
+        try:
+            self._reader, self._plugin = self._get_reader(
+                image,
+                reader,
+                use_plugin_cache,
+                fs_kwargs,
+                **kwargs,
+            )
+        except biob.exceptions.UnsupportedFileFormatError:
+            # When reading from S3 if we failed trying to read
+            # try reading as an anonymous user otherwise re-raise
+            # the error
+            if not str(image).startswith("s3://"):
+                raise
+
+            self._reader, self._plugin = self._get_reader(
+                image,
+                reader,
+                use_plugin_cache,
+                fs_kwargs | {"anon": True},
```

Comment:
the implication here, then, is that if you passed fs_kwargs={anon:True} in the initial bioImage constructor then s3 reads do work?

Reply:
Yeah 👍 (and I believe were we to always passed `True` here then users that need to auth like for private buckets would fail, haven't tested that explicitly yet)